src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java

```diff
@@ -29,6 +30,10 @@
 public class PetClinicApplication {
 
 	public static void main(String[] args) {
+		// enable Application Insights Java programmatically, see
+		// https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-spring-boot#enabling-programmatically
+		ApplicationInsights.attach();
```

Comment:
There's a recent change in App Service Java17 container that breaks programmatic attach. For the time being, you'll need to add this change in the `infra` folder for `app-service`: https://github.com/Azure/azure-dev/pull/1217/files

Reply:
Might be solved if you just use the unreleased version of [api-appservice-java.bicep](https://github.com/Azure/azure-dev/pull/1217/files#diff-b4dd64ba60ffa7851a99f4256a0cba8dc3ce5af6f113d0e39070a9fbc88f7769)

Reply:
I ran into this error a while ago, but didn't know how to fix it at that time, thanks a lot!